opendatahub-io/kserve
# add oauth-proxy to rawdeployments if odh auth label is present

**What this PR does / why we need it**: 

This PR adds the following : 
- If the isvc has label `"security.opendatahub.io/enable-auth" = "true"` 
-- an oauth proxy container is added to the deployment 
-- http port is replaced by https port in the service 
-- `"service.beta.openshift.io/serving-cert-secret-name"` is added to the service to allow creation of the tls secret 
- The service account, CRB and Route are created in https://github.com/opendatahub-io/odh-model-controller/pull/274 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://issues.redhat.com/browse/RHOAIENG-10291, https://issues.redhat.com/browse/RHOAIENG-13444 
TEST WITH: https://github.com/opendatahub-io/odh-model-controller/pull/274

**Type of changes**
Please delete options that are not relevant.


**Feature/Issue validation/testing**:

- Deploy ODH/RHOAI:
- Install the DSC (kserve spec below) : 
```
    kserve:
      defaultDeploymentMode: RawDeployment
      devFlags:
        manifests:
          - contextDir: config
            sourcePath: overlays/odh
            uri: 'https://github.com/VedantMahabaleshwarkar/kserve/tarball/devflags'
          - contextDir: config
            sourcePath: ''
            uri: 'https://github.com/VedantMahabaleshwarkar/odh-model-controller/tarball/devflags'
      managementState: Managed
      serving:
        ingressGateway:
          certificate:
            type: OpenshiftDefaultIngress
        managementState: Managed
        name: knative-serving
```

- To test enabling route for a particular ISVC 
--  Create any kserve isvc+SR
-- isvc should have annotation : `"serving.kserve.io/deploymentMode" : RawDeployment`
-- isvc should have label : `"networking.kserve.io/visibility": "enable-route"`


- To test route with auth: 
--  Create any kserve isvc+SR
-- isvc should have annotation : `"serving.kserve.io/deploymentMode" : RawDeployment`
-- isvc should have label: `"security.opendatahub.io/enable-auth" = "true"` 
-- isvc should have label: `"networking.kserve.io/visibility": "enable-route"`

- To test Inference without auth: 
-- remove isvc label `"security.opendatahub.io/enable-auth" = "true"`  

Now inference should work without token 

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:


**Release note**:

```release-note
Added oauth-proxy authentication for Kserve RawDeployment Routes
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.
